src/coord/src/coord.rs

```diff
@@ -3039,27 +3046,18 @@ impl Coordinator {
         let make_sink_desc = |coord: &mut Coordinator, from, from_desc, uses| {
             // Determine the frontier of updates to tail *from*.
             // Updates greater or equal to this frontier will be produced.
-            let frontier = if let Some(ts) = ts {
-                // If a timestamp was explicitly requested, use that.
-                let id_bundle = coord.catalog.nearest_indexes(uses);
-                let ts = coord.determine_timestamp(
-                    session,
-                    &id_bundle,
-                    PeekWhen::AtTimestamp(ts),
-                    compute_instance,
-                )?;
-                Antichain::from_elem(ts)
-            } else {
-                coord.determine_frontier(uses, compute_instance)
-            };
+            let id_bundle = coord.catalog.nearest_indexes(uses);
+            // If a timestamp was explicitly requested, use that.
+            let timestamp =
+                coord.determine_timestamp(session, &id_bundle, when, compute_instance)?;
```

Comment:
I think I might prefer this being 'coord.least_valid_read()` instead? I guess there is a legit question about whether TAIL should present as much or as little historical context that it has. Counter-argument: commands that do not have `AS OF` in them should present as sequentially ordered. It's a pity choosing such a late time for e.g. constant queries though.